Python 3.12 support

tsdownsample doesn't install with a Python 3.12 environment (tested locally with 3.12.0 and in Github CI)

However, I've been able to build it and pass the Pytest tests after switching Rust to `rustup default nightly` and removing the pins on the dependencies in Cargo.toml (it wouldn't build before these steps).


```
Building wheels for collected packages: tsdownsample
  Building wheel for tsdownsample (pyproject.toml): started
  Building wheel for tsdownsample (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  × Building wheel for tsdownsample (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [100 lines of output]
      Running `maturin pep517 build-wheel -i /opt/hostedtoolcache/Python/3.12.0/x64/bin/python --compatibility off`
      📦 Including license file "/tmp/pip-install-d4u9kedm/tsdownsample_64a0754dc6ce4c4daaf39edb4c31f838/LICENSE"
      ⚠️ Warning: specify [package.metadata.maturin] name in Cargo.toml is deprecated, use module-name in [tool.maturin] section in pyproject.toml instead
      🍹 Building a mixed python/rust project
      🔗 Found pyo3 bindings
      🐍 Found CPython 3.12 at /opt/hostedtoolcache/Python/3.12.0/x64/bin/python
      📡 Using build options bindings from pyproject.toml
         Compiling autocfg v1.1.0
         Compiling cfg-if v1.0.0
Failed to build tsdownsample
         Compiling libc v0.2.142
         Compiling target-lexicon v0.12.6
         Compiling memoffset v0.8.0
         Compiling crossbeam-utils v0.8.15
         Compiling scopeguard v1.1.0
         Compiling libm v0.2.6
         Compiling pyo3-build-config v0.18.3
         Compiling num-traits v0.2.15
         Compiling crossbeam-epoch v0.9.14
         Compiling once_cell v1.17.1
         Compiling proc-macro2 v1.0.56
         Compiling rayon-core v1.11.0
         Compiling unicode-ident v1.0.8
         Compiling quote v1.0.26
         Compiling crossbeam-deque v0.8.3
         Compiling crossbeam-channel v0.5.8
         Compiling num_cpus v1.15.0
         Compiling num-integer v0.1.45
         Compiling syn v1.0.109
         Compiling pyo3-ffi v0.18.3
         Compiling lock_api v0.4.9
         Compiling either v1.8.1
         Compiling parking_lot_core v0.9.7
         Compiling rawpointer v0.2.1
         Compiling matrixmultiply v0.3.3
        |            ^^^^^^^
      
      
      error[E0554]: `#![feature]` may not be used on the stable release channel
       --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/argminmax-0.3.1/src/lib.rs:4:12
        |
      4 | #![feature(int_roundings)]
        |            ^^^^^^^^^^^^^
      
      
      error: aborting due to 6 previous errors
      
      
      For more information about this error, try `rustc --explain E0554`.
      
      Error: command ['maturin', 'pep517', 'build-wheel', '-i', '/opt/hostedtoolcache/Python/3.12.0/x64/bin/python', '--compatibility', 'off'] returned non-zero exit status 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for tsdownsample
ERROR: Could not build wheels for tsdownsample, which is required to install pyproject.toml-based projects
Error: Process completed with exit code 1.
```

--- a/.github/workflows/ci-tsdownsample.yml
+++ b/.github/workflows/ci-tsdownsample.yml
@@ -46,7 +46,7 @@ jobs:
       matrix:
         os: ['windows-latest', 'macOS-latest', 'ubuntu-latest']
         rust: ['nightly']  # ['stable', 'beta']
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11', '3.12']
     
     env:
       PYTHON: ${{ matrix.python-version }}

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -21,6 +21,7 @@ classifiers = [
     'Programming Language :: Python :: 3.9',
     'Programming Language :: Python :: 3.10',
     'Programming Language :: Python :: 3.11',
+    'Programming Language :: Python :: 3.12',
     'Operating System :: POSIX',
     'Operating System :: MacOS :: MacOS X',
     'Operating System :: Microsoft :: Windows'

